server/src/modules/rules/helpers/rule.comparator.service.ts

```diff
@@ -500,26 +502,48 @@ export class RuleComparatorService {
     }
 
     if (action === RulePossibility.BEFORE) {
+      assertIsDate(val1);
+      assertIsDate(val2);
       return val1 && val2 ? val1 <= val2 : false;
     }
 
     if (action === RulePossibility.AFTER) {
+      assertIsDate(val1);
+      assertIsDate(val2);
       return val1 && val2 ? val1 >= val2 : false;
     }
 
     if (action === RulePossibility.IN_LAST) {
       return (
-        val1 >= val2 && // time in s
+        (val1 as Date) >= val2 && // time in s
         (val1 as unknown as Date) <= new Date()
       );
     }
 
     if (action === RulePossibility.IN_NEXT) {
       return (
-        val1 <= val2 && //  time in s
+        (val1 as Date) <= val2 && //  time in s
         (val1 as unknown as Date) >= new Date()
       );
     }
+
+    if (action === RulePossibility.COUNT_EQUALS) {
+      assertIsArray(val1);
+      return val1.length === val2;
+    }
+    if (action === RulePossibility.COUNT_NOT_EQUALS) {
+      return !this.doRuleAction(val1, val2, RulePossibility.COUNT_EQUALS);
+    }
+    if (action === RulePossibility.COUNT_BIGGER) {
+      assertIsArray(val1);
+      assertIsNumber(val2);
+      return val1.length > val2;
+    }
+    if (action === RulePossibility.COUNT_SMALLER) {
+      assertIsArray(val1);
+      assertIsNumber(val2);
+      return val1.length < val2;
+    }
   }
```

Comment:
It's possible to select a [list] for val2 so you'll want something like this `(Array.isArray(val2) ? val2.length : val)`